modules/osdk-operatorconditions.adoc

```diff
@@ -0,0 +1,67 @@
+= Enabling Operator conditions
+
+Operator Lifecycle Manager (OLM) provides Operators with a channel to communicate complex states that influence its behavior while managing the Operator. By default, OLM creates an `OperatorCondition` custom resource definition (CRD) when it installs an Operator; based on the conditions set in the `OperatorCondition` custom resource (CR), OLM changes its behavior accordingly.
+
+To support Operator conditions, an Operator must be able to read the `OperatorCondition` CR created by OLM and have the ability to:
+
+* Get the specific condition.
+* Set the status of a specific condition.
+
+This can be accomplished using the link:https://github.com/operator-framework/operator-lib/tree/v0.3.0[`operator-lib`] library. An Operator author can provide a link:https://github.com/kubernetes-sigs/controller-runtime/tree/master/pkg/client[`controller-runtime` client] in their Operator, which the library can use to access the `OperatorCondition` CR owned by the Operator in the cluster.
+
+The library provides a generic `Conditions` interface, which has the following methods to `Get` and `Set` a `conditionType` in the `OperatorCondition` CR:
+
+`Get`:: To get the specific condition, the library uses the `client.Get` function from `controller-runtime`, which requires an `ObjectKey` of type `types.NamespacedName` present in `conditionAccessor`.
+
+`Set`:: To update the status of the specific condition, the library uses the `client.Update` function from `controller-runtime`. An error occurs if the `conditionType` is not present in the CRD.
+
+The Operator is only allowed to modify the `status` subresource of the CR. Operators can either delete or update the `status.conditions` array to include the condition. For more details on the format and description of the fields present in the conditions, see the upstream link:https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#Condition[Condition GoDocs].
+
+[NOTE]
+====
+Operator SDK v1.4.0 supports `operator-lib` v0.3.0.
+====
+
+.Prerequisites
+
+* An Operator project generated using the Operator SDK.
+
+.Procedure
+
+. In the `go.mod` file of your Operator project, add `operator-framework/operator-lib` as a required library:
++
+[source,go]
+----
+module github.com/example-inc/memcached-operator
+
+go 1.15
+
+require (
+  k8s.io/apimachinery v0.19.2
+  k8s.io/client-go v0.19.2
+  sigs.k8s.io/controller-runtime v0.7.0
+  operator-framework/operator-lib v0.3.0
+)
+----
+
+. Write your own constructor in your Operator logic that:
++
+--
+* Accepts a `controller-runtime` client.
+* Accepts a `conditionType`.
+* Returns a `Condition` interface to update or add conditions.
+--
++
+For example, because OLM currently supports the `Upgradeable` condition, creating an interface which has methods to access the `Upgradeable` condition can be as simple as:
++
+[source,go]
+----
+func NewUpgradeable(cl client.Client) (Condition, error) {
+	return NewCondition(cl, "Upgradeable")
+}
```

Comment:
yeah, this looks right. This can further be used as:
```go
con, err := NewUpgradable(cl); //cl is the client here
```
`cond` would in turn have `Get` and `Set` methods, like `cond.Get(context)`.